src/Microsoft.Health.Fhir.Shared.Core/Features/Resources/BaseConditionalHandler.cs

```diff
@@ -54,5 +60,66 @@ protected async Task<SearchResultEntry[]> Search(string instanceType, IReadOnlyL
 
             return matchedResults;
         }
+
+        public async Task ResolveBundleReferencesAsync(Resource resource, Dictionary<string, (string resourceId, string resourceType)> referenceIdDictionary, string requestUrl, CancellationToken cancellationToken)
```

Comment:
Similarly., not sure if this would be the correct location for this method. Should this be in a separate class?

Reply:
Since we need to resolve references for conditional resource whether transaction, bundle or observation So I think it makes sense to have it on this class. Thoughts?

Reply:
yeah, the method name confused me. We can change the name of method to something that's not only pointing to bundle, since its resolving reference within othertype of resources as well?